### DISCUSSION TITLE
How to access this.$page in Vue 3 composition API

### DISCUSSION THREAD

**Discussion by sadhakbj** [inertiajs/inertia#781] [Help]
I am new to Inertia js, i am following the official documentation and PINGCRM's code to learn it. I want to make use of Vue 3 Composition API. In my Layout.vue file, i have the following code to check if the url is active:

```js
export default defineComponent({
    name: "navbar",
    setup() {
        const isUrl = (...urls) => {
            return urls.filter((url) => this.$page.url.match(url)).length
        }
        return { isUrl }
    },
})
```

While trying to do so i am getting the following error: 
`app.js:22025 Uncaught (in promise) TypeError: Cannot read property '$page' of undefined`

I looked in the official documentation but was unable to find the solution for this.

**Comment by reinink** [comment]
Use the `usePage` hook. For example:

```vue
<script>
  import { computed } from 'vue'
  import { usePage } from '@inertiajs/inertia-vue3'

  export default {
    setup() {
      const url = usePage().url
    },
  }
</script>
```

**Reply by sadhakbj** [reply]
Thank you it helped.

**Reply by mreduar** [reply]
Hi @reinink 
I am trying to use this hook in Vue 3 with `<script setup>` 
trying to recreate `$page.props.user.current_team.name`
But for some reason the user does not come in `usePage().props`
Please could you give me a hint on how to access the logged in user information using composition api? Thanks

**Reply by ajnsn** [reply]
Hey @mreduar 
First off I would either use the [Vue Dev Tools](https://devtools.vuejs.org/) (for v3 you need the v6.x, currently beta) or the network tab in your Browser Dev Tools (e.g. [Chrome](https://developer.chrome.com/docs/devtools/network/), [Firefox](https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/request_list)) to see if the prop is really filled with desired data.

You could also log the props object to your console, like so:

```vue
<script setup>
import { usePage } from '@inertiajs/inertia-vue3';

const pageProps = usePage().props;
console.log({ ...pageProps.value });
</script>
```

I am pretty sure that your `user` prop might be missing or the name/structure is different. This might be because [you are not sharing](https://inertiajs.com/shared-data) it correctly or even did not [register the middleware](https://inertiajs.com/server-side-setup#middleware) (when using Laravel). 

Hope this helps!

**Reply by mreduar** [reply]
Hi @ajnsn, thanks for your reply.
I am using the Vue Dev Tools in Chrome, and the `user` property is present here.
![image](https://user-images.githubusercontent.com/22550119/145584365-8a5962c3-0aab-4743-ba98-287247291443.png)

But when I use `usePage().props` it is not. It tells me undefined
![image](https://user-images.githubusercontent.com/22550119/145584626-c91a0fd6-97f9-4078-ab21-a92f8ba5b8cd.png)
![image](https://user-images.githubusercontent.com/22550119/145584725-6badd697-c16e-4599-b62d-7b787abbf8dc.png)

That's why I thought that `usePage()` is not the same as `$page`.

**Reply by saas786** [reply]
> ```
> .value
> ```

You missed the value part should be like `usePage().props.value.user`

**Reply by mreduar** [reply]
Ohhhh, I thought it was just one example, I didn't think I actually had to access `.value`. Indeed there if this property. Thank you very much @saas786 and @ajnsn

**Reply by ajnsn** [reply]
Hey @mreduar 
This should work:
```js
console.log('pageProps', pageProps.value.user);
```

The reason behind the `.value` is that the props are an `computed()` property, see source [over here](https://github.com/inertiajs/inertia/blob/master/packages/inertia-vue3/src/app.js#L100).

Vue 3 always returns a reactive object that has an "inner value".
And here are some more insights from the [Vue docs](https://v3.vuejs.org/guide/reactivity-computed-watchers.html#computed-values).

Note: When using in a `<template>`, you can stick to `pageProps.user` as Vue will automatically unwrap the value for you.

**Reply by mreduar** [reply]
@ajnsn Great! Thanks for these documentations 👏👏👏 I still have to learn the new things vue 3
